euphoria-spark/src/main/java/cz/seznam/euphoria/spark/SparkExecutor.java

```diff
@@ -66,16 +67,31 @@ public void shutdown() {
     submitExecutor.shutdownNow();
   }
 
+  /**
+   * Set accumulator provider that will be used to collect metrics and counters.
+   * When no provider is set a default instance of {@link VoidAccumulatorProvider}
+   * will be used.
+   *
+   * @param factory Factory to create an instance of accumulator provider.
+   */
+  public void setAccumulatorProvider(SparkAccumulatorFactory factory) {
+    this.accumulatorFactory = Objects.requireNonNull(factory);
+  }
+
   @Override
   public void setAccumulatorProvider(AccumulatorProvider.Factory factory) {
-    this.accumulatorFactory = Objects.requireNonNull(factory);
+    this.accumulatorFactory = new SparkAccumulatorFactory.Adapter(
+            Objects.requireNonNull(factory));
   }
 
   private Result execute(Flow flow) {
     if (!isBoundedInput(flow)) {
       throw new UnsupportedOperationException("Spark executor doesn't support unbounded input");
     }
 
+    // init accumulators
+    accumulatorFactory.init(sparkContext);
```

Comment:
That is good point :+1: I haven't considered this use-case at first, my bad.
Seems the `accumulatorFactory` needs to be cloned for each flow and this shouldn't be a problem since the factory is serializable anyway. Please have a look at the newly added commit if it makes sense.